fe/fe-core/src/main/java/org/apache/doris/analysis/AggregateInfo.java

```diff
@@ -283,6 +285,12 @@ public static boolean estimateIfContainsMultiDistinct(List<FunctionCallExpr> dis
                 hasMultiDistinct = true;
             }
         }
+
+        // for vectorized execution, we force it to using hash set to execution
+        if (!hasMultiDistinct && distinctAggExprs.size() == 1 && distinctAggExprs.get(0).getFnParams().isDistinct()
+                && VectorizedUtil.isVectorized() && ConnectContext.get().getSessionVariable().enableSingleDistinctColumnOpt()) {
+            hasMultiDistinct = true;
```

Comment:
nit: if we use 2 stage aggregate for one distinct aggregate, maybe we should change function name and variable name to get a better description of what's actually being done